### ISSUE TITLE
Training ate my dataframe

### ISSUE THREAD

**Issue by calz1** [ClimbsRocks/auto_ml#214]
I'm on the latest version from pip as of this morning. It seems a little funny, but the shape of my training dataframe changes after train() runs.

Here's before:
```
df_train, df_test = train_test_split(df, test_size=0.2, random_state=42)
print(df_train.shape)
df_train2=df_train
```

which prints:

> (1153, 2)

I'm trying with a simple dataset that is basically a comment field and a target of 0 or 1. Next I have this which completes fine:

```
ml_predictor = Predictor(type_of_estimator='classifier', column_descriptions=column_descriptions )
ml_predictor.train(df_train,model_names='XGBClassifier')
ml_predictor.score(df_test, df_test.pkscore)
```
and I get information on the brier loss score as expected. But when I run it against the training set, I get this:

> 
> ---------------------------------------------------------------------------
> AttributeError                            Traceback (most recent call last)
> <ipython-input-6-9cb5b0a2aad5> in <module>()
>       3 
>       4 #Score on training data to ensure we are not overfitting
> ----> 5 ml_predictor.score(df_train, df_train.pkscore)
> 
> /usr/lib64/python3.6/site-packages/pandas/core/generic.py in __getattr__(self, name)
>    2968             if name in self._info_axis:
>    2969                 return self[name]
> -> 2970             return object.__getattribute__(self, name)
>    2971 
>    2972     def __setattr__(self, name, value):
> 
> AttributeError: 'DataFrame' object has no attribute 'pkscore'

pkscore being my target column.  The error message is correct, because this:

`print(df_train.shape)`

now yields:

> (1153, 1)

It appears to be a problem in train() and not in score() because if I make a copy of df_train before running train(), this works fine:

`ml_predictor.score(df_train2, df_train2.pkscore)`

**Comment by ClimbsRocks** [comment]
Hahah, great issue name, and the behavior you noted definitely exists. 

One thing i'm optimizing for with auto_ml is not using memory unnecessarily. It's far from perfect, but I've made some design decisions to try to allow people to load datasets that are as large as possible onto any hardware, not just giant cloud instances with hundreds of gigs of RAM. 

I made a lazy decision at one point in this pursuit. Your issue helped me see just how lazy this was and how relatively easy it is to work around it. 

I think the latest commit should have fixed it. I just added a new test to verify that's running now, then assuming that goes well i'll deploy to pip this afternoon. 

Thanks for your continued issues!
